docs/en/sql-reference/functions/tuple-map-functions.md

```diff
@@ -0,0 +1,38 @@
+---
+toc_priority: 46
+toc_title: Working with maps
+---
+
+# Functions for maps {#functions-for-working-with-tuple-maps}
+
+## mapAdd(Tuple(Array, Array), Tuple(Array, Array) [, ...]) {#function-mapadd}
+
+Collect all the keys, sum corresponding values and return one map with summed data.
+All key types should be integer and same.
```

Comment:
Why do you limit key type to integers only? It seems `String` support will be the top feature request after mapAdd is available in release.

Reply:
I wasn't sure that this patch will be merged, for now it has exact features that we need in our company.

Reply:
In the last commit I have added support of other key types (String, enums, dates and UUIDs). I haven't added Nullable support since `sumMap` also doesn't support them and this functionality will just decrease the current performance.